kobo-uncaged/device/device.go

```diff
@@ -481,24 +484,77 @@ func (k *Kobo) SaveCoverImage(contentID string, size image.Point, imgB64 string)
 	}
 }
 
-// UpdateNickelDB updates the Nickel database with updated metadata obtained from a previous run
+// UpdateNickelDB updates the Nickel database with updated metadata obtained from a previous run,
+// or this run if updating via triggers
 func (k *Kobo) UpdateNickelDB() error {
-	// No matter what happens, we remove the 'metadata_update.kobouc' file when we're done
-	defer os.Remove(filepath.Join(k.BKRootDir, kuUpdatedMDfile))
-	query := `
+	if !k.KuConfig.AddMetadataByTrigger {
```

Comment:
It would be a good idea to use transactions here, as it's significantly faster and a bit safer.

Reply:
Good idea. Will look into this when I get home later.